client/me/purchases/confirm-cancel-purchase/index.jsx

```diff
@@ -24,43 +24,36 @@ const ConfirmCancelPurchase = React.createClass( {
 		selectedSite: React.PropTypes.object
 	},
 
-	componentWillMount() {
-		recordPageView( 'confirm_cancel_purchase', this.props );
+	getInitialState() {
+		return {
+			html: null
+		};
 	},
 
-	componentWillReceiveProps( nextProps ) {
-		recordPageView( 'confirm_cancel_purchase', this.props, nextProps );
+	componentWillMount() {
+		this.loadEndpointForm();
+		recordPageView( 'confirm_cancel_purchase', this.props );
 	},
 
-	componentDidMount() {
-		if ( ! this.props.selectedPurchase.hasLoadedFromServer || ! this.props.selectedSite ) {
-			return;
+	componentDidUpdate( prevProps, prevState ) {
+		if ( ! prevState.html && this.state.html ) {
+			const container = React.findDOMNode( this.refs.root );
+
+			container.innerHTML = this.state.html;
```

Comment:
Most like we could use `dangerouslySetInnerHTML` but I used another approach pointed in the following comment :)